docs/RFCS/postgres_syntax.md

```diff
@@ -0,0 +1,98 @@
+- Feature Name: Postgres Syntax Compatibility
+- Status: draft
+- Start Date: 2017-04-16
+- Authors: Matt Jibson
+- RFC PR:
+- Cockroach Issue:
+
+# Summary
+
+This proposal is to define the specific goals of the two syntax types
+(traditional and modern) so we can correctly decide how to triage parsing
+differences between CockroachDB and Postgres, Spanner, or other databases.
+
+# Motivation
+
+The differences in syntax types (`traditional` and `modern`) in CockroachDB
+are not documented and have no defined goals or reasons for existing or
+differing. The unwritten reasons are that traditional is the old Postgres
+style and modern is the new Spanner style. However, we have already
+made some choices to differ from Postgres ([integer division returning
+decimals](https://github.com/cockroachdb/cockroach/pull/3308), for example)
+because we thought the Postgres syntax could be improved. This style worked
+fine for a while.
+
+With our recent goal of supporting various ORMs, these previous
+decisions become more complicated. We are advertising ourselves
+as an implementer of the Postgres protocol and syntax, and we
+mostly do. There is some syntax we do not yet support, and we
+rightly return errors for that. However we now have various ([integer
+division](https://github.com/cockroachdb/cockroach/pull/3308), [bitwise xor
+(~) operator](https://github.com/cockroachdb/cockroach/pull/14908), [shift
+argument modulo](https://github.com/cockroachdb/cockroach/issues/14874)) query
+expressions that are known to return different results that Postgres. In all
+of those cases we decided to differ because we believe the Postgres syntax
+to be flawed.
```

Comment:
This entire paragraph mistakes syntax for semantics.

At this point I think there is a general confusion  in the text about the opposition between "traditional" vs. "modern"  **syntax**, and "pg", "spanner", and "cockroachdb" **semantics**.

I am going to read the rest of the text as a discussion of semantic differences. If that is what you wanted to convey then I suggest renaming the RFC and keep the words "traditional" and "modern" out of the picture at the beginning, and only outline in the end that perhaps our desire to distinguish **dialects** (a word which we aren't using so far, but see my comments below) should not be implemented with a feature (eg "`set syntax = ...`") that carries the term "syntax" -- as this word (and the associations it creates in the user's mind) would be mistakes.